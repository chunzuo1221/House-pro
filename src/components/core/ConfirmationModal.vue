<template>
    <div class="confirmation-modal">
        <div class="modal-content">
            <div class="text-center" v-html="message"></div>
        </div>
        <div class="modal-footer">
            <button @click="onConfirm">Yes</button>
            <button @click="onCancel">No</button>
        </div>
    </div>
</template>

<script>

export default {
  name: 'ConfirmationModal',
  props: ['message'],
  methods: {
    onConfirm () {
      this.$emit('confirmed')
    },
    onCancel () {
      this.$emit('canceled')
    }
  }
}

</script>

<style scoped>

.modal-enter {
  opacity: 0;
}
.confirmation-modal .modal-content div {
  padding-top: 30px;
  padding-bottom: 20px;
}
.confirmation-modal button {
  padding:  10px 40px;
  border: 1px solid black;
}
.confirmation-modal button:hover,
.confirmation-modal button:focus {
  border: none;
}
.confirmation-modal .modal-content {
  box-shadow: none;
  padding: 0px 20px;
}
.confirmation-modal .modal-footer {
  padding-top: 20px;
  padding-bottom: 20px;
}

</style>
