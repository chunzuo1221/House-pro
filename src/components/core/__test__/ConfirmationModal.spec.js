import { shallowMount } from '@vue/test-utils'
import ConfirmationModal from '@/components/core/ConfirmationModal'

describe('ConfirmationModal', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(ConfirmationModal)
    expect(wrapper.element).toMatchSnapshot()
  })
})
