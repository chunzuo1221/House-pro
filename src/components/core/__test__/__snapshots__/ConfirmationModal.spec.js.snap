// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModal renders correctly 1`] = `
<div
  class="confirmation-modal"
>
  <div
    class="modal-content"
  >
    <div
      class="text-center"
    />
  </div>
   
  <div
    class="modal-footer"
  >
    <button>
      Yes
    </button>
     
    <button>
      No
    </button>
  </div>
</div>
`;
